FE/src/components/SingleRecipe.jsx

```diff
@@ -1,87 +1,573 @@
+                    "Authorization" :`BEARER ${token}`
+                },
+            });
+            const json = await response.json();
+            return json;
+        } catch (error) {
+            setDeleteCommentErrMess(commentId);
+            throw (error);
+        }
+    }
+
+    const potentialDeletedComment= await deleteCommentFetch(commentId);
+    if(potentialDeletedComment){
+        setCommentAreYouSure(null);
+        setDeleteCommentErrMess(null);
+        setRefreshCounter((prev) => prev + 1);
+    } else {
+        setDeleteCommentErrMess(commentId);
+    }
+}
+
+async function handleDeleteRecipe(recipeId){
+    async function deleteRecipeFetch(recipeId){
+        try {
+            const response = await fetch(`http://localhost:3000/api/recipes/${recipeId}`,
+            { 
+                method: "DELETE",
+                headers: { 
+                    "Content-Type": "application/json",
+                    "Authorization" :`BEARER ${token}`
+                },
+            });
+            const json = await response.json();
+            return json;
+        } catch (error) {
+            setDeleteRecipeErrMess(true);
+            throw (error);
+        }
+    }
+
+    const potentialDeletedRecipe= await deleteRecipeFetch(recipeId);
+    if(potentialDeletedRecipe){
+        setRecipeAreYouSure(false);
+        setDeleteRecipeErrMess(false);
+        setRefreshCounter((prev) => prev + 1);
+    } else {
+        setDeleteRecipeErrMess(true);
+    }
+}
+
 
     return (
         <>
+        {(errMess || !recipe.id) ?
+        <p>There has been an error</p>
+        :
         <div className="singleRecipeCard">
             <h1>{recipe.title}</h1>
-            <h5>@{recipe.userName}</h5>
-            <div>{tags.map((tag) => {
+            <h5>@{recipe.user.username}</h5>
+            {admin &&
+            <>
+                <button onClick={()=>navigate(`/recipes/${recipe.id}/edit`)}>Edit Recipe</button>
+                {recipeAreYouSure ?
+                    <>
+                        <p>Are you are you want to delete this?</p>
+                        <button onClick={()=>handleDeleteRecipe(recipe.id)}>Yes</button>
+                        <button onClick={()=>setRecipeAreYouSure(false)}>No</button>
+                    </>
+                :
+                    <>
+                        <button onClick={()=>setRecipeAreYouSure(true)}>Delete Recipe</button>
+                    </>
+                }
+                {deleteRecipeErrMess && <p>There has been an error deleting the recipe.</p>}
+            </>
+            }
+            <div>{recipe.tags.map((tag) => {
                 return (
-                <p key={tag}><em>{tag}</em></p>
+                <p key={tag.id}><em>{tag.name}</em></p>
             )})}</div>
-            <img src={recipe.imgUrl} heigth="20%" width="25%"/>
-            <h3>Ingredients:</h3>
+            <img src={recipe.imgurl} width="500px"/>
```

Comment:
Remove the width from here. I'm handling the width of all of the images in the CSS, and have asked all of you to remove the width attributes because it screws up the CSS grid.